main.js

```diff
@@ -118,4 +134,53 @@ document.addEventListener("DOMContentLoaded", function () {
         break;
     }
   }
+
+  // Edit a to-do item
+  function editTask(index) {
+    const newText = prompt("Edit task", tasks[index].text);
+    if (newText !== null) {
+      tasks[index].text = newText;
+      createNewTodoItem();
+      saveTasks();
+    }
+  }
+
+  // Removing and editing tasks
+  todoList.addEventListener("click", (event) => {
+    const index = event.target.dataset.index;
+    if (event.target.classList.contains("remove")) {
+      removeTask(index);
+    } else if (event.target.classList.contains("edit")) {
+      editTask(index);
+    }
+  });
+
+  // Save tasks to localStorage
+  function saveTasks() {
+    localStorage.setItem("tasks", JSON.stringify(tasks));
```

Comment:
try to save only new changes, now you're saving all object

Reply:
How we can save only changed item? Finally we must to save "tasks" array in local storage. Would be good if you suggest some examples. Thanks!